testing-common/src/main/groovy/io/opentelemetry/instrumentation/test/base/AbstractHttpServerTest.groovy

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.test.base
+
+import ch.qos.logback.classic.Level
+import io.opentelemetry.instrumentation.test.AgentTestRunner
+import io.opentelemetry.instrumentation.test.utils.OkHttpUtils
+import io.opentelemetry.instrumentation.test.utils.PortUtils
+import okhttp3.OkHttpClient
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import spock.lang.Shared
+
+abstract class AbstractHttpServerTest<SERVER> extends AgentTestRunner {
```

Comment:
Instead of an abstract class, consider making it a trait (like `ApplicationConfigTrait` or `InstrumentationTestTrait`. `withRetryOnAddressAlreadyInUse` could possibly be moved to `PortUtils` to remove dependency on `AgentTestRunner` (`AgentTestRunner` does not seem to be a proper place for this method anyway)

Reply:
Moved code from abstract class to HttpServerTestTrait, also moved retry logic to RetryOnAddressAlreadyInUseTrait